### DISCUSSION TITLE
browserBlock extension block

### DISCUSSION THREAD

**Discussion by Bucknalla** [balenablocks/blocks#164] [Ideas]
A block that sends REST requests based upon button input detection, would be super useful for something like a browserblock.

For example GPIO 1 [pressed] -> `curl --data "url=www.balena.io" http://localhost:5011/url` so you can navigate a kiosk without having to mess around with a URL bar or a touchscreen.

**Comment by tmigone** [comment]
I like this! We can have a user provided config file to know what GPIO to take over and what the URLs and payloads are.
This would enable building sound boards for example quite easily, i'm sure there is a use case for balenaSound

**Reply by Bucknalla** [reply]
@tmigone yeah definitely! I like the idea of being able to quickly build a physical UI with a couple of blocks mashed together.
